### ISSUE TITLE
The kubectl annotate documentation for the overwrite option is incorrect

### ISSUE THREAD

**Issue by testworksau** [kubernetes/kubernetes#111375]
### What happened?

Relying on [the annotate documentation](https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#annotate), I attempted to overwrite an annotation using:

```
kubectl annotate --overwrite pv $PV_NAME my.annotation/should.update=yes
```

The command failed to run and reported:

```
error: --overwrite is false but found the following declared annotation(s): 'my.annotation/should.update' already has a value (annotated)
```

`--overwrite` defaults to false, so we actually need to pass in `--overwrite=true` for this to work. 

### What did you expect to happen?

The command should have succeeded, according to the documentation

### How can we reproduce it (as minimally and precisely as possible)?

1. Add an annotation to a resource
2. Attempt to overwrite the annotation by passing in `--overwrite` and not `--overwrite=true`

### Kubernetes version

<details>
Irrelevant
</details>

### Cloud provider

<details>
Irrelevant
</details>

**Comment by testworksau** [comment]
Just waiting on CLA approval to occur (needs to go through our legal team, so may take a little while).

**Comment by brianpursley** [comment]
This works for me:
```
$ kubectl run foo --image=nginx
pod/foo created
$ kubectl annotate pod foo bar=baz
pod/foo annotated
$ kubectl annotate pod foo bar=changed
error: --overwrite is false but found the following declared annotation(s): 'bar' already has a value (baz)
$ kubectl annotate pod foo bar=changed --overwrite
pod/foo annotated
```

I didn't have to set `--overwrite=true`.  Just `--overwrite` seemed to work fine.  And this is how all the other boolean flags work; the presence of the flag itself is the same as true.

It is strange that you got the error you did.

First of all, is my example above the same thing you are trying?

Second, what version of kubectl are you using.  Can you post the output of `kubectl version`?

**Comment by testworksau** [comment]
Hi @brianpursley - thanks for looking into this.

I have no idea why, but I cannot reproduce this anymore. I can specify `--overwrite` without explicitly supplying `=true`  and the annotation in my example now works.

I took the commands in my description from my build system logs, which had a few annotate steps, so it's possible I simply missed setting `--overwrite` on one of the entries (sadly I rebased my commits so can no longer tell).

At least, that seems to be the most plausible reason given your explanation above. 

I must admit, I did think it was weird that specifying `--overwrite` would default to false, but I am new to kubectl and the documentation that is getting spat out of the CLI is a little odd to me:

```
    --overwrite=false:
        If true, allow annotations to be overwritten, otherwise reject annotation updates that overwrite existing
        annotations.
```
